src/content/scrolling.ts

```diff
@@ -166,12 +167,23 @@ export async function recursiveScroll(
     if (!node) {
         const sameSignX = xDistance < 0 === lastX < 0
         const sameSignY = yDistance < 0 === lastY < 0
-        if (lastRecursiveScrolled && sameSignX && sameSignY) {
+        const sameElement = lastFocused == document.activeElement
+        if (lastRecursiveScrolled && sameSignX && sameSignY && sameElement) {
             // We're scrolling in the same direction as the previous time so
             // let's try to pick up from where we left
             startingFromCached = true
             node = lastRecursiveScrolled
         } else {
+
+            // Try scrolling the active node or one of its parent elements
+            node = document.activeElement
+            while (true) {
+                if ((await scroll(xDistance, yDistance, node))) return true
+                node = node.parentElement
+                if (node === null) break
```

Comment:
Can `node` really be `null` here? I thought it'd be `Element | undefined`.

Reply:
It's `null` for the parent of the root html element, for real. I don't think you can have active orphaned elements but maybe you are right and we should use == undefined so people's computers don't take off because HTML doesn't work like I think it does